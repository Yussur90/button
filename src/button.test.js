import React from "react";
import { render, cleanup, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import "jest-styled-components";
import Button from "./button";

afterEach(cleanup);

describe("button test", () => {
  test("the button is rendered correctly", () => {
    const { asFragment } = render(<Button />);
    expect(asFragment()).toMatchSnapshot();
  });

  test("button called function on click", () => {
    const handleClick = jest.fn();

    const { getByTestId } = render(<Button onClick={handleClick}></Button>);

    expect(getByTestId("button")).toBeTruthy();
    expect(getByTestId("button")).toHaveProperty("click");
    fireEvent.click(getByTestId("button"));
    expect(handleClick).toHaveBeenCalled();
  });

  test("loading element is rendered and have progress props", () => {
    const { getByTestId, rerender } = render(
      <Button loading={true} progress={100} />
    );

    const loadingElement = getByTestId("loading");

    expect(loadingElement).toBeInTheDocument();

    expect(loadingElement).toHaveAttribute("progress");
    rerender(<Button loading={false} />);
    expect(loadingElement).not.toBeInTheDocument();
  });

  test("background color of the button is correct", () => {
    const { getByTestId } = render(<Button />);
    expect(getByTestId("button")).toHaveStyle(`
  background-color: "#504de5"`);
  });

  test("color of the button is correct", () => {
    const { getByTestId } = render(<Button />);
    expect(getByTestId("button")).toHaveStyle(`color: "white"`);
  });

  test("button have disabled attribute", () => {
    const { getByTestId } = render(<Button />);
    expect(getByTestId("button")).toHaveProperty("disabled");
  });

  test("button have type attribute", () => {
    const { getByTestId } = render(<Button />);
    expect(getByTestId("button")).toHaveProperty("type");
  });

  test("button have style attribute", () => {
    const { getByTestId } = render(<Button />);
    expect(getByTestId("button")).toHaveProperty("style");
  });

  test("children for button", () => {
    const { container } = render(<Button />);

    expect(container.children.length).toBe(1);
  });
});
