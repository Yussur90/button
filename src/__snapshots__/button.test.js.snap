// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button test the button is rendered correctly 1`] = `
<DocumentFragment>
  <button
    class="css-xnktn1"
    data-testid="button"
    type="submit"
  />
</DocumentFragment>
`;
